python/pysrc/kaskada/_timestream.py

```diff
@@ -59,11 +59,19 @@ def __init__(self, ffi: _ffi.Expr) -> None:
     @staticmethod
     def _literal(value: Literal, session: _ffi.Session) -> Timestream:
         """Construct a Timestream for a literal value."""
-        if isinstance(value, timedelta):
-            raise TypeError("Cannot create a literal Timestream from a timedelta")
-        elif isinstance(value, datetime):
+        if isinstance(value, datetime):
             raise TypeError("Cannot create a literal Timestream from a datetime")
-        return Timestream(_ffi.Expr.literal(session, value))
+
+        if isinstance(value, timedelta):
+            # The smallest unit stored in timedelta is us
+            us = value / timedelta(microseconds=1)
+            # Get the seconds as an integer
+            seconds = int(us / 1_000_000)
+            # Get the leftover nanoseconds
+            nanoseconds = int((us % 1_000_000) * 1_000)
+            return Timestream(_ffi.Expr.literal_datetime(session, seconds, nanoseconds))
```

Comment:
I _could_ just convert to pure nanos here, which would avoid the math in sparrow conversion. Thoughts? `Timedelta` can only represent down to `us`, so we don't have that nanosecond granularity, but that's a different consideration anyway.

Reply:
I can see either option making sense. Probably OK to keep the split representation for now (main loss to standardizing on nanos is not being able to represent deltas or date times greater than about 292 years.